File: api/pkg/db/models/runtime.go
package models

import (
	"time"
)

type Runtimes struct {
	Name         string           `gorm:"primaryKey" json:"name"`
	Versions     []RuntimeVersion `gorm:"foreignKey:RuntimeName" json:"versions"`
	Image        string           `json:"image"`
	BuildCommand string           `json:"build_command"`
	StartCommand string           `json:"start_command"`
	CreatedAt    time.Time        `gorm:"autoCreateTime;column:created_at" json:"createdAt"`
	UpdatedAt    time.Time        `gorm:"autoCreateTime;column:updated_at" json:"updatedAt"`
}

File: apps/web/src/features/services/types/Runtime.ts
export interface IRuntime {
    name: string;
    image: string;
    versions: IRuntimeVersion[];
    build_command: string;
    start_command: string;
}

export interface IRuntimeVersion {
    id: string;
    runtime_name: string;
    name: string;
    tag: string;
}